hcloud/client.go

```diff
@@ -371,7 +371,10 @@ func errorFromResponse(resp *http.Response, body []byte) error {
 	if respBody.Error.Code == "" && respBody.Error.Message == "" {
 		return nil
 	}
-	return ErrorFromSchema(respBody.Error)
+
+	hcErr := ErrorFromSchema(respBody.Error)
+	hcErr.HTTPStatusCode = resp.StatusCode
+	return hcErr
```

Comment:
Oh, this one fell off my radar.

I am not sure if it is smart to diverge from the API schema here, in particular if we use a field.

What do you think about adding a `hcErr.Response()` method that returns the embedded `hcErr.response` value.

Reply:
Nice suggestion, makes it more versatile.

I have updated your commit to work with our own `http.Response` wrapper.